Validation/HGCalValidation/src/TICLCandidateValidator.cc

```diff
@@ -0,0 +1,735 @@
+        "num_track_cand_vs_pt_" + charged[i], charged[i] + " track efficiency vs pT;p_{T} (GeV)", 50, 0, 250));
+    h_num_chg_pt_candidate_pdgId.push_back(ibook.book1D(
+        "num_pid_cand_vs_pt_" + charged[i], charged[i] + " track and PID efficiency vs pT;p_{T} (GeV)", 50, 0, 250));
+    h_num_chg_pt_candidate_energy.push_back(
+        ibook.book1D("num_energy_cand_vs_pt_" + charged[i],
+                     charged[i] + " track, PID and energy efficiency vs pT;p_{T} (GeV)",
+                     50,
+                     0,
+                     250));
+    h_den_chg_eta_candidate.push_back(
+        ibook.book1D("den_cand_vs_eta_" + charged[i], charged[i] + " simCandidates eta;#eta (GeV)", 50, -3, 3));
+    h_num_chg_eta_candidate_track.push_back(ibook.book1D(
+        "num_track_cand_vs_eta_" + charged[i], charged[i] + " track efficiency vs eta;#eta (GeV)", 50, -3, 3));
+    h_num_chg_eta_candidate_pdgId.push_back(ibook.book1D(
+        "num_pid_cand_vs_eta_" + charged[i], charged[i] + " track and PID efficiency vs eta;#eta (GeV)", 50, -3, 3));
+    h_num_chg_eta_candidate_energy.push_back(
+        ibook.book1D("num_energy_cand_vs_eta_" + charged[i],
+                     charged[i] + " track, PID and energy efficiency vs eta;#eta (GeV)",
+                     50,
+                     -3,
+                     3));
+    h_den_chg_phi_candidate.push_back(ibook.book1D(
+        "den_cand_vs_phi_" + charged[i], charged[i] + " simCandidates phi;#phi (GeV)", 50, -3.14159, 3.14159));
+    h_num_chg_phi_candidate_track.push_back(ibook.book1D("num_track_cand_vs_phi_" + charged[i],
+                                                         charged[i] + " track efficiency vs phi;#phi (GeV)",
+                                                         50,
+                                                         -3.14159,
+                                                         3.14159));
+    h_num_chg_phi_candidate_pdgId.push_back(ibook.book1D("num_pid_cand_vs_phi_" + charged[i],
+                                                         charged[i] + " track and PID efficiency vs phi;#phi (GeV)",
+                                                         50,
+                                                         -3.14159,
+                                                         3.14159));
+    h_num_chg_phi_candidate_energy.push_back(
+        ibook.book1D("num_energy_cand_vs_phi_" + charged[i],
+                     charged[i] + " track, PID and energy efficiency vs phi;#phi (GeV)",
+                     50,
+                     -3.14159,
+                     3.14159));
+  }
+}
+
+void TICLCandidateValidator::fillCandidateHistos(const edm::Event& event,
+                                                 edm::Handle<ticl::TracksterCollection> simTrackstersCP_h) {
+  edm::Handle<std::vector<TICLCandidate>> TICLCandidates_h;
+  event.getByToken(TICLCandidatesToken_, TICLCandidates_h);
+  auto TICLCandidates = *TICLCandidates_h;
```

Comment:
Could be just
```suggestion
  auto TICLCandidates = event.get(TICLCandidatesToken_);
```
Is the copy intentional? (if not, `auto` -> `auto const&`)

Reply:
switched to `auto`